torchvision/prototype/models/vision_transformer.py

```diff
@@ -443,3 +380,74 @@ def vit_l_32(*, weights: Optional[ViT_L_32_Weights] = None, progress: bool = Tru
         progress=progress,
         **kwargs,
     )
+
+
+def interpolate_embeddings(image_size: int, patch_size: int, model_state: Mapping[str, torch.Tensor], reset_heads=False):
+    """This function helps interpolating positional embeddings during checkpoint loading,
+    especially when you want to apply a pre-trained model on images with different resolution.
+
+    Returns a model state dict which can be loaded into the new model.
+
+    Args:
+        image_size (int): Image size of the new model.
+        patch_size (int): Patch size of the new model.
+        model_state (Mapping[str, torch.Tensor]): State dict of the pre-trained model.
+        reset_heads (bool): If true, not copying the state of heads. Default: False.
+    """
```

Comment:
Also the return on the documentation.

Reply:
Hey would you let to check the latest version? I've addressed these issues you mentioned in the last few commits :)